lib/services/kubernetes.go

```diff
@@ -176,32 +175,32 @@ func UnmarshalKubeCluster(data []byte, opts ...MarshalOption) (types.KubeCluster
 	return nil, trace.BadParameter("unsupported kube cluster resource version %q", h.Version)
 }
 
-const (
-	// labelTeleportKubeClusterName is the label key containing the kubernetes cluster name override.
-	labelTeleportKubeClusterName = types.TeleportNamespace + "/kubernetes-name"
-)
-
-// setKubeName modifies the types.Metadata argument in place, setting the kubernetes cluster name.
-// The name is calculated based on nameParts arguments which are joined by hyphens "-".
-// If the kube_cluster name override label is present (setKubeName), it will replace the *first* name part.
-func setKubeName(meta types.Metadata, firstNamePart string, extraNameParts ...string) types.Metadata {
-	nameParts := append([]string{firstNamePart}, extraNameParts...)
-
-	// apply override
-	if override, found := meta.Labels[labelTeleportKubeClusterName]; found && override != "" {
-		nameParts[0] = override
-	}
+// setAWSKubeName modifies the types.Metadata in place, overriding the first
+// part if the kube cluster override label for AWS is present, and setting the
+// kube cluster name.
+func setAWSKubeName(meta types.Metadata, firstNamePart string, extraNameParts ...string) types.Metadata {
+	return setResourceName(types.AWSKubeClusterNameOverrideLabel, meta, firstNamePart, extraNameParts...)
```

Comment:
For AWS can we just support both cases to avoid having users memorizing the different options depending on the cloud provider they use?

Reply:
so remove `teleport.dev/kubernetes-name` and just have all clouds use `TeleportKubernetesName`?

Reply:
Support both actually

Reply:
added for both db and kube clusters in AWS here: [0c01dc0](https://github.com/gravitational/teleport/pull/28917/commits/0c01dc0ce7302c4b992ac617a39d2827f9405df7)

The extra diff is enormous because I had to fix a bunch of test cases for databases, and I added tests for kube clusters.